content/automate/ManageIQ/System/CommonMethods/MiqAe.class/__methods__/weightedupdatestatus.rb

```diff
@@ -29,6 +29,20 @@ def reconcile_states_percent(path, states)
               percent
             end
 
+            def create_cleanup_request(task)
+              sm_uri = @handle.object['cleanup_state_machine']
+              if sm_uri.present? && task.get_option(:cleanup_request_id).absent?
+                options = {}
+                sm_uri_array = sm_uri.split('/')
+                options[:instance_name] = sm_uri_array.pop
+                options[:class_name] = sm_uri_array.pop
+                options[:namespace] = sm_uri_array.join('/')
+                options[:user_id] = @handle.root['user_id']
+                request = @handle.execute("create_automation_request", options, 'admin', true)
```

Comment:
can we get current user instead of default to 'admin'?

Reply:
@fdupont-redhat Does the user exist in the DB?
https://github.com/ManageIQ/manageiq-automation_engine/blob/bbcf066a1ebb87fc7021baff6189a32db0054765/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb#L109
Its finding the user based on the User#userid field not the User.id field.
If we have a current user in the automation we should be able to use it. $evm.root['user'].userid

Reply:
Yes, the user exists. I have created a user named `bob` with id 2. I have tried the following in rails console:

```ruby
$evm.execute("create_automation_request", { instance_name: "Test", class_name: "Raise", namespace: "StateMachines", user_id: 2 }, 'bob', true)
```

And the error is the following:

```ruby
RuntimeError: not authorized
	from /home/miq/manageiq/app/models/miq_approval.rb:66:in `user_validate'
	from /home/miq/manageiq/app/models/miq_approval.rb:14:in `approve'
	from /home/miq/manageiq/app/models/miq_request.rb:307:in `approve'
	from /home/miq/manageiq/app/models/mixins/inter_region_api_method_relay.rb:30:in `block (2 levels) in api_relay_method'
	from /home/miq/manageiq/app/models/miq_request.rb:521:in `post_create'
	from /home/miq/manageiq/app/models/miq_request.rb:494:in `create_request'
	from /home/miq/manageiq/app/models/mixins/inter_region_api_method_relay.rb:48:in `block (2 levels) in api_relay_class_method'
	from /home/miq/.gem/ruby/bundler/gems/manageiq-automation_engine-bbcf066a1ebb/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb:110:in `create_automation_request'
	from /home/miq/.gem/ruby/bundler/gems/manageiq-automation_engine-bbcf066a1ebb/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service.rb:167:in `execute'
	from (irb):6
	from /home/miq/.gem/ruby/gems/railties-5.0.7/lib/rails/commands/console.rb:65:in `start'
	from /home/miq/.gem/ruby/gems/railties-5.0.7/lib/rails/commands/console_helper.rb:9:in `start'
	from /home/miq/.gem/ruby/gems/railties-5.0.7/lib/rails/commands/commands_tasks.rb:78:in `console'
	from /home/miq/.gem/ruby/gems/railties-5.0.7/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
	from /home/miq/.gem/ruby/gems/railties-5.0.7/lib/rails/commands.rb:18:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'
```

Reply:
After @gmcculloug comment, I did some tests around approval and it appears that the role the user belongs to needs the Services > Requests > Operate > Approve and Deny privilege, whereas I would expect it needs Automation > Automate > Requests > Operate > Approve and Deny, as it is an automation request and not a service provision request. Because, it means we have to document why the AppOwner role requires Service > Requests > Operate > Approve and Deny privilege.

Anyway, I do agree that it should be the current user that is the requester, instead of admin. But we have to document RBAC for 5.9.4, because RBAC for Transformation is not available yet.